matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/pushers/DefaultPushersService.kt

```diff
@@ -66,27 +66,39 @@ internal class DefaultPushersService @Inject constructor(
                                deviceDisplayName: String,
                                url: String,
                                append: Boolean,
-                               withEventIdOnly: Boolean)
-            : UUID {
-        // Do some parameter checks. It's ok to throw Exception, to inform developer of the problem
-        if (pushkey.length > 512) throw InvalidParameterException("pushkey should not exceed 512 chars")
-        if (appId.length > 64) throw InvalidParameterException("appId should not exceed 64 chars")
-        if ("/_matrix/push/v1/notify" !in url) throw InvalidParameterException("url should contain '/_matrix/push/v1/notify'")
-
-        val pusher = JsonPusher(
-                pushKey = pushkey,
-                kind = "http",
-                appId = appId,
-                appDisplayName = appDisplayName,
-                deviceDisplayName = deviceDisplayName,
-                profileTag = profileTag,
-                lang = lang,
-                data = JsonPusherData(url, EVENT_ID_ONLY.takeIf { withEventIdOnly }),
-                append = append)
+                               withEventIdOnly: Boolean
+    ) = addPusher(
+            JsonPusher(
+                    pushKey = pushkey,
+                    kind = "http",
+                    appId = appId,
+                    profileTag = profileTag,
+                    lang = lang,
+                    appDisplayName = appDisplayName,
+                    deviceDisplayName = deviceDisplayName,
+                    data = JsonPusherData(url, EVENT_ID_ONLY.takeIf { withEventIdOnly }),
+                    append = append
+            )
+    )
 
-        val params = AddHttpPusherWorker.Params(sessionId, pusher)
+    override fun addEmailPusher(email: String, lang: String, emailBranding: String, appDisplayName: String, deviceDisplayName: String) = addPusher(
+            JsonPusher(
+                    pushKey = email,
+                    kind = "email",
+                    appId = "m.email",
```

Comment:
Maybe create public const val for `kind` "email" and "http" (so in `PusherService`) and private const `appId` "m.email" ?

Reply:
sounds good, will do :+1: